src/libraries/System.Private.CoreLib/src/System/Reflection/MethodBase.cs

```diff
@@ -220,6 +220,8 @@ BindingFlags invokeAttr
 
                             isValueType = sigType.CheckValue(ref arg, ref copyBackArg, binder, culture, invokeAttr);
                         }
+
+                        copyBackArg = sigType.IsNullableOfT ? ParameterCopyBackAction.CopyNullable : ParameterCopyBackAction.Copy;
```

Comment:
I think it would be better to write-back the default value right here, and not depend on the copy back that is done after the invoke.

You will need to change the parameters argument to from `ReadOnlySpan<object>` to `object[]` to make that work.

Reply:
> I think it would be better to write-back the default value right here,

I think we should write back the default value before conversions (before the CheckValue call). It will better match the earlier behavior where the default value that was written back was also without any conversions.

Reply:
Let me give it try